src/argilla/client/users.py

```diff
@@ -0,0 +1,118 @@
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import warnings
+from datetime import datetime
+from typing import TYPE_CHECKING, List, Optional, Union
+from uuid import UUID
+
+from pydantic import ValidationError
+
+import argilla.client.sdk.users.api as users_api
+from argilla.client import active_client
+from argilla.client.sdk.commons.errors import AlreadyExistsApiError, BaseClientError
+from argilla.server.models import UserRole as Roles
+
+if TYPE_CHECKING:
+    from argilla.client.sdk.users.models import UserModel
+
+
+class User:
+    username: str
+    id: UUID
+    first_name: str
+    last_name: Optional[str]
+    role: Roles
+    workspaces: Optional[List[str]]
+    api_key: str
+    inserted_at: datetime
+    updated_at: datetime
+
+    def __init__(self, username: Optional[str] = None, *, id: Optional[str] = None) -> None:
+        assert username or id, "Either the `username` or `id` must be provided."
+
+        existing_user = self.__user_exists(username=username, id=id)
+        if existing_user is None:
+            raise ValueError(
+                f"User with username=`{username}` and/or id=`{id}` doesn't exist in Argilla, so please create it first via the `User.create` method."
+            )
+        self.__dict__.update(existing_user.dict())
+
+    @staticmethod
+    def __user_exists(username: Optional[str] = None, id: Optional[str] = None) -> Optional["UserModel"]:
+        # loop through `list_users` and look for matches on either `username` or `id`, if
+        # the user exists return the reponse, otherwise return None
+        for user in list_users():
+            if user.username == username or user.id == id:
+                return user
+        return None
+
+    def __str__(self) -> str:
+        return f"User(id={self.id}, username={self.username}, role={self.role}, workspaces={self.workspaces}, api_key={self.api_key}, first_name={self.first_name}, last_name={self.last_name}, role={self.role}, inserted_at={self.inserted_at}, updated_at={self.updated_at})"
+
+    def delete(self) -> None:
+        users_api.delete_user(active_client().http_client.httpx, self.id)
+
+    @classmethod
+    def create(
+        cls,
+        username: str,
+        password: Union[str, UUID],
+        first_name: Optional[str] = None,
+        last_name: Optional[str] = None,
+        role: Optional[Roles] = None,
+    ) -> "User":
+        if not first_name:
+            warnings.warn(
```

Comment:
No it does not, but we decided to use both `logger.warning` and `warnings.warn`, as those are intended to different purposes, see https://docs.python.org/3/howto/logging.html#when-to-use-logging

<img width="658" alt="image" src="https://github.com/argilla-io/argilla/assets/36760800/489747ac-8a29-4544-97eb-7963d0f4ccae">

Reply:
In this case, defining a `first_name` makes the warning disappear, so this is why here we're using `warnings.warn` instead of `logging.warning`, anyway we can have a more in-detail discussing for this matter! 👍🏻